### PR TITLE
Some more wavm cleanup

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/wavm.cpp [hunk 1]
```diff
@@ -194,6 +194,35 @@ namespace wavm_host_module {
   };
 } // namespace wavm_host_module
 
+
+void WavmEngine::verifyContract(vector<uint8_t> const& code)
```
Comment (axic) [inline src/wavm.cpp]: @chfast I'm not sure how to do this without code duplication given the mess `GCPointer` is causing. Any ideas?

Comment (chfast) [reply src/wavm.cpp]: What exactly is duplicated?

Comment (axic) [reply src/wavm.cpp]: This entire body below is present in `internalExecute`.

**Comment by axic** [comment]
@chfast it can be reviewed now :wink:

This should conclude the wavm work, everything works now.

File: src/wavm.cpp [hunk 2]
```diff
@@ -393,4 +394,91 @@ ExecutionResult WavmEngine::internalExecute(
...
+
+    ensureCondition(
+      eei_signatures.count(import.exportName),
+      ContractValidationFailure,
+      "Importing invalid EEI method."
+    );
+    IR::FunctionType const& eei_function_type = eei_signatures.at(import.exportName);
+
+    ensureCondition(
+      moduleIR.types.size() >= import.type.index,
```
Comment (axic) [inline src/wavm.cpp]: @chfast is indexing of a vector going from 0 or 1? `moduleIR.types` is a vector.

Comment (chfast) [reply src/wavm.cpp]: From 0.

**Pull request merged**